runtime/compiler/runtime/J9CodeCache.cpp

```diff
@@ -134,8 +145,48 @@ J9::CodeCache::initialize(TR::CodeCacheManager *manager,
 
    if (!self()->OMR::CodeCache::initialize(manager, codeCacheSegment, allocatedCodeCacheSizeInBytes))
       return false;
+
    self()->setInitialAllocationPointers();
 
+#ifdef LINUX
+   if (manager->isDisclaimEnabled())
+      {
+      J9JavaVM * javaVM = jitConfig->javaVM;
+      PORT_ACCESS_FROM_JAVAVM(javaVM); // for j9vmem_supported_page_sizes
+
+      uint8_t *middle  = _warmCodeAlloc + (_coldCodeAllocBase - _warmCodeAlloc) / 2;
+      size_t round = j9vmem_supported_page_sizes()[0] - 1;
+      middle = (uint8_t *)(((size_t)(middle + round)) & ~round);
+
+      TR_ASSERT_FATAL(_coldCodeAlloc > middle, "A code cache can't be smaller than a page");
+
+      size_t coldCacheSize = _coldCodeAlloc - middle;
+      coldCacheSize = (coldCacheSize + round) & ~round;
+
+      if (madvise(middle, coldCacheSize, MADV_NOHUGEPAGE) != 0)
+         {
+         const char *error = strerror(errno);
+         if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance))
+            TR_VerboseLog::writeLineLocked(TR_Vlog_INFO, "Failed to set MADV_NOHUGEPAGE for code cache: %s: %p %zu", error, middle, coldCacheSize);
+         }
+      else if (TR::Options::getCmdLineOptions()->getVerboseOption(TR_VerbosePerformance))
+         {
+         TR_VerboseLog::writeLineLocked(TR_Vlog_INFO, "In code cache %p small pages start from %p\n", this, middle);
```

Comment:
I would prefer this message to be more explicit. I am trying to imagine what a casual consumer of verbose logs would understand when it sees this message among all the other others. 
Suggestion: "Forcing code cache cold region %p-%p to use default size memory pages"